django/db/backends/base/creation.py

```diff
@@ -130,8 +131,15 @@ def deserialize_db_from_string(self, data):
         the serialize_db_to_string() method.
         """
         data = StringIO(data)
-        for obj in serializers.deserialize("json", data, using=self.connection.alias):
-            obj.save()
+        # Run without constraint checks (which requires a transaction), so any forward references or even loops in the
+        # data can be loaded without problems.
+        with atomic(using=self.connection.alias):
+            with self.connection.constraint_checks_disabled():
```

Comment:
SQLite cannot disable constraints within a transaction

https://github.com/django/django/blob/c33eb6dcd0c211f8f02b2976fe3b3463f0a54498/django/db/backends/sqlite3/base.py#L289-L296

So you'll likely need another solution here.

Reply:
Looks like it works at the start of a transaction, right?

Reply:
It has to be issued before a transaction is opened. In short `'PRAGMA foreign_keys = OFF'` is a silent noop after a `BEGIN` has been issued.

Reply:
More details about it from https://sqlite.org/pragma.html

Reply:
It happens to work here because all base backends except for MySQL use deferred foreign keys (MySQL doesn't support them) so no checks are performed until the transaction commits.

Reply:
Oh, interesting, then `loaddata` also has it in the wrong order. I wonder if MySQL, which also uses this method, can have the order swapped or not... Also, I think I have not seen test failures on Sqlite even with this order, so maybe sqlite already defers constraint checks to the end of a transaction and offers `disable_constraint_checking` to disable constraint checking outside of a transaction? This needs some additional investigation and testing, I guess.

Reply:
So, for this particular case, it is ok that `disable_constraint_checking()` fails inside a transaction, since the transaction itself already provides the needed semantics. I guess this does mean that the API of `disable_constraint_checking()` and `constraint_checks_disabled()` is a bit ill-defined, but maybe that is out of scope for this PR.

So I would propose keeping the code as-is, since it should work for the purpose? @charettes, would you agree?

Reply:
@matthijskooijman lets keep it as is if `loaddata` uses this pattern. I guess we could eventually add a connection level database backend specified manager that hides this implementation detail but this should do in the mean time.